doc/CloudIntegration.xml

```diff
@@ -0,0 +1,338 @@
+        based on Web Services and OpenAPI. The service provider (MCS) implements the ONVIF service.
+        The service is described using the YAML-based file. Then, the YAML file is used as the basis
+        for the service requester (OCS) implementation/integration. Integration is simplified
+        through the use of code generating tools that generate platform specific code that can be
+        used by the developers to integrate the Web Service.</para>
+      <para>The Web Service provider and requester communicate using the OpenAPI message exchange
+        protocol. OpenAPI is a lightweight messaging protocol used to encode the information in a
+        Web Service request and in a response message before sending them over a network. OpenAPI
+        messages are independent of any operating system or protocol and may be transported using a
+        variety of Internet protocols. This ONVIF standard defines conformant transport protocols
+        for the SOAP messages for the described Web Services.</para>
+      <para>The Web Service overview section introduces into the general ONVIF service structure,
+        the command definition syntax in the specification, error handling principles and the
+        adopted Web Service security mechanisms.</para>
+      </section>
+      <section xml:id="_Ref208379139">
+        <title>Security</title>
+        <section>
+          <title>Authentication</title>
+          <para>The services defined in this standard, whenever consumed, shall be protected using
+          digest authentication according to [RFC 2617] over HTTP and HTTPS or JWT client
+          authorization based on [RFC 6750], only over HTTPS. Additional information about HTTP
+          digest  authenticationcan can be found the ONVIF Core specification, while the JWT client
+          authorization can be found in the ONVIF Security Service.</para>
+          <para>MCS and OCS may support the SHA-256 hashing algorithm in HTTP and HTTPS. An ONVIF
+          compliant service should follow [RFC 7616].</para>
+        </section>
+        <section>
+          <title>User levels and access classes for the service</title>
+          <para>Since this service is only meant to enable sharing devices between different cloud
+          services, there is no need for defining different access classes to authorized access to
+          the functions defined in this specification.</para>
+        </section>
+      </section>
+    <section xml:id="_Ref208379149">
+      <title>Format of the share token</title>
+      <para>To start the device sharing operation, the OCS must provide a share token to the PCS, to
+        prove it has the rights to access the camera. They way the OCS retrieves the share token is
+        beyond the scope of this specification.</para>
+      <para>The share token must be a JWT and must include the <emphasis>xaddr</emphasis> claim that
```

Comment:
I'm wondering if we'd like to also mandate an identifying claim related to the device in there. So that a OCS can get a rough idea of what it expects to receive.
I'm thinking the Serial Number of the device maybe, as defined in GetDeviceInformation (Core spec). This can guide feedback from the OCS (identifying information that can be shown in the UI), and can also guide the naming of the ClientId for the AuthorizationServer info

Reply:
Another possibility is to use the different URIs for different cameras. Imagine a schema like `https://your.host/callback/identifier` where the callback handler is always the same and by looking at the `identifier` it determines the camera. Again, I am open to discuss.

Reply:
That's way before the callback though. I'd like to have a unique identifier for the device _before_ attempting to upgrade.
For example, I need to create a ClientId for this device in my identity provider. I'd like to have the serial of the device, or maybe mac address, in there, so that the device identity is tied to the device in some way.
Right now, the token has nothing identifying the device at all. So I need to perform the onboard to completion before I'm able to find out anything uniquely identifying about it (via the Uplink)

Reply:
As per my comment above @jmelancongen I agree with this it needs more clear work flows defined.

Reply:
@jmelancongen let's add more claims to the JWT. I am not a VMS expert, what do you think the JWT should include. I can understand the serial number, I can also probably understand the model. But I do not understand why you are looking for the MAC address. the MAC address is "link local" the cloud VMS will never see it.